<!DOCTYPE html>
<html >

<head>

  <meta charset="UTF-8">
  <meta http-equiv="X-UA-Compatible" content="IE=edge">
  <title>6 Interface and Usage | GRID: Greenfield Image Decoder</title>
  <meta name="description" content="A Python Package of Greenfield Image Decoder for Aerial High-Throughput Phenotyping">
  <meta name="generator" content="bookdown  and GitBook 2.6.7">

  <meta property="og:title" content="6 Interface and Usage | GRID: Greenfield Image Decoder />
  <meta property="og:type" content="book" />
  <meta property="og:url" content="https://github.com/poissonfish/photo_grid" />
  <meta property="og:image" content="https://github.com/poissonfish/photo_gridmd/icon.png" />
  <meta property="og:description" content="A Python Package of Greenfield Image Decoder for Aerial High-Throughput Phenotyping" />
  <meta name="github-repo" content="poissonfish/photo_grid" />

  <meta name="twitter:card" content="summary" />
  <meta name="twitter:title" content="6 Interface and Usage | GRID: Greenfield Image Decoder />
  
  <meta name="twitter:description" content="A Python Package of Greenfield Image Decoder for Aerial High-Throughput Phenotyping" />
  <meta name="twitter:image" content="https://github.com/poissonfish/photo_gridmd/icon.png" />

<meta name="author" content="Chunpeng James Chen">


<meta name="date" content="2019-07-29">

  <meta name="viewport" content="width=device-width, initial-scale=1">
  <meta name="apple-mobile-web-app-capable" content="yes">
  <meta name="apple-mobile-web-app-status-bar-style" content="black">
  
  
<link rel="prev" href="get-started-1.html">

<script src="libs/jquery-2.2.3/jquery.min.js"></script>
<link href="libs/gitbook-2.6.7/css/style.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-table.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-bookdown.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-highlight.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-search.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-fontsettings.css" rel="stylesheet" />







<script src="libs/kePrint-0.0.1/kePrint.js"></script>


<style type="text/css">
a.sourceLine { display: inline-block; line-height: 1.25; }
a.sourceLine { pointer-events: none; color: inherit; text-decoration: inherit; }
a.sourceLine:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode { white-space: pre; position: relative; }
div.sourceCode { margin: 1em 0; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
code.sourceCode { white-space: pre-wrap; }
a.sourceLine { text-indent: -1em; padding-left: 1em; }
}
pre.numberSource a.sourceLine
  { position: relative; left: -4em; }
pre.numberSource a.sourceLine::before
  { content: attr(title);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; pointer-events: all; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
    color: #aaaaaa;
  }
pre.numberSource { margin-left: 3em; border-left: 1px solid #aaaaaa;  padding-left: 4px; }
div.sourceCode
  {  }
@media screen {
a.sourceLine::before { text-decoration: underline; }
}
code span.al { color: #ff0000; font-weight: bold; } /* Alert */
code span.an { color: #60a0b0; font-weight: bold; font-style: italic; } /* Annotation */
code span.at { color: #7d9029; } /* Attribute */
code span.bn { color: #40a070; } /* BaseN */
code span.bu { } /* BuiltIn */
code span.cf { color: #007020; font-weight: bold; } /* ControlFlow */
code span.ch { color: #4070a0; } /* Char */
code span.cn { color: #880000; } /* Constant */
code span.co { color: #60a0b0; font-style: italic; } /* Comment */
code span.cv { color: #60a0b0; font-weight: bold; font-style: italic; } /* CommentVar */
code span.do { color: #ba2121; font-style: italic; } /* Documentation */
code span.dt { color: #902000; } /* DataType */
code span.dv { color: #40a070; } /* DecVal */
code span.er { color: #ff0000; font-weight: bold; } /* Error */
code span.ex { } /* Extension */
code span.fl { color: #40a070; } /* Float */
code span.fu { color: #06287e; } /* Function */
code span.im { } /* Import */
code span.in { color: #60a0b0; font-weight: bold; font-style: italic; } /* Information */
code span.kw { color: #007020; font-weight: bold; } /* Keyword */
code span.op { color: #666666; } /* Operator */
code span.ot { color: #007020; } /* Other */
code span.pp { color: #bc7a00; } /* Preprocessor */
code span.sc { color: #4070a0; } /* SpecialChar */
code span.ss { color: #bb6688; } /* SpecialString */
code span.st { color: #4070a0; } /* String */
code span.va { color: #19177c; } /* Variable */
code span.vs { color: #4070a0; } /* VerbatimString */
code span.wa { color: #60a0b0; font-weight: bold; font-style: italic; } /* Warning */
</style>

</head>

<body>



  <div class="book without-animation with-summary font-size-2 font-family-1" data-basepath=".">

    <div class="book-summary">
      <nav role="navigation">

<ul class="summary">
<li class="chapter" data-level="" data-path="index.html"><a href="index.html"><i class="fa fa-check"></i>Why GRID?</a><ul>
<li class="chapter" data-level="0.0.1" data-path="index.html"><a href="index.html#get-started"><i class="fa fa-check"></i><b>0.0.1</b> Get Started</a></li>
</ul></li>
<li class="chapter" data-level="1" data-path="interface-and-usage.html"><a href="interface-and-usage.html"><i class="fa fa-check"></i><b>1</b> Interface and Usage</a><ul>
<li class="chapter" data-level="1.1" data-path="interface-and-usage.html"><a href="interface-and-usage.html#overview-of-grid"><i class="fa fa-check"></i><b>1.1</b> Overview of GRID</a><ul>
<li class="chapter" data-level="" data-path="interface-and-usage.html"><a href="interface-and-usage.html#load-data"><i class="fa fa-check"></i>Load data</a></li>
<li class="chapter" data-level="" data-path="interface-and-usage.html"><a href="interface-and-usage.html#crop-your-image"><i class="fa fa-check"></i>Crop your image</a></li>
<li class="chapter" data-level="" data-path="interface-and-usage.html"><a href="interface-and-usage.html#plots-extraction"><i class="fa fa-check"></i>Plots extraction</a></li>
<li class="chapter" data-level="" data-path="interface-and-usage.html"><a href="interface-and-usage.html#set-up-center-points"><i class="fa fa-check"></i>Set up center points</a></li>
<li class="chapter" data-level="" data-path="interface-and-usage.html"><a href="interface-and-usage.html#finalize-plots-boundary"><i class="fa fa-check"></i>Finalize plot’s boundary</a></li>
</ul></li>
<li class="chapter" data-level="1.2" data-path="interface-and-usage.html"><a href="interface-and-usage.html#load-data-1"><i class="fa fa-check"></i><b>1.2</b> Load data</a></li>
<li class="chapter" data-level="1.3" data-path="interface-and-usage.html"><a href="interface-and-usage.html#crop-images"><i class="fa fa-check"></i><b>1.3</b> Crop Images</a></li>
<li class="chapter" data-level="1.4" data-path="interface-and-usage.html"><a href="interface-and-usage.html#plots-extraction-1"><i class="fa fa-check"></i><b>1.4</b> Plots extraction</a></li>
</ul></li>
<li class="chapter" data-level="2" data-path="load-data-2.html"><a href="load-data-2.html"><i class="fa fa-check"></i><b>2</b> Load data</a></li>
<li class="chapter" data-level="3" data-path="step-by-step-tutorial.html"><a href="step-by-step-tutorial.html"><i class="fa fa-check"></i><b>3</b> Step by Step Tutorial</a><ul>
<li class="chapter" data-level="3.1" data-path="step-by-step-tutorial.html"><a href="step-by-step-tutorial.html#gwas-in-gapit"><i class="fa fa-check"></i><b>3.1</b> GWAS in GAPIT</a><ul>
<li><a href="step-by-step-tutorial.html#prepare-plink-files"><strong><em>Prepare PLINK files</em></strong></a></li>
<li><a href="step-by-step-tutorial.html#import-files"><strong><em>Import files</em></strong></a></li>
<li><a href="step-by-step-tutorial.html#define-gapit"><strong><em>Define GAPIT</em></strong></a></li>
<li><a href="step-by-step-tutorial.html#inspect-results"><strong><em>Inspect results</em></strong></a></li>
</ul></li>
<li class="chapter" data-level="3.2" data-path="step-by-step-tutorial.html"><a href="step-by-step-tutorial.html#k-fold-cross-validation-gs-in-bglr"><i class="fa fa-check"></i><b>3.2</b> K-fold cross-validation GS in BGLR</a><ul>
<li><a href="step-by-step-tutorial.html#prepare-vcf-files"><strong><em>Prepare VCF files</em></strong></a></li>
<li><a href="step-by-step-tutorial.html#import-files-1"><strong><em>Import files</em></strong></a></li>
<li><a href="step-by-step-tutorial.html#define-bglr"><strong><em>Define BGLR</em></strong></a></li>
<li><a href="step-by-step-tutorial.html#inspect-results-1"><strong><em>Inspect results</em></strong></a></li>
</ul></li>
<li class="chapter" data-level="3.3" data-path="step-by-step-tutorial.html"><a href="step-by-step-tutorial.html#gwas-assisted-gs-in-rrblup"><i class="fa fa-check"></i><b>3.3</b> GWAS-assisted GS in rrBLUP</a><ul>
<li><a href="step-by-step-tutorial.html#import-files-2"><strong><em>Import files</em></strong></a></li>
<li><a href="step-by-step-tutorial.html#define-farmcpu-gwas"><strong><em>Define FarmCPU (GWAS)</em></strong></a></li>
<li><a href="step-by-step-tutorial.html#define-rrblup-gs"><strong><em>Define rrBLUP (GS)</em></strong></a></li>
<li><a href="step-by-step-tutorial.html#inspect-progress"><strong><em>Inspect Progress</em></strong></a></li>
</ul></li>
<li class="chapter" data-level="3.4" data-path="step-by-step-tutorial.html"><a href="step-by-step-tutorial.html#convert-hapmap-to-plink-format"><i class="fa fa-check"></i><b>3.4</b> Convert Hapmap to PLINK format</a><ul>
<li><a href="step-by-step-tutorial.html#prepare-hapmap-files"><strong><em>Prepare Hapmap files</em></strong></a></li>
<li><a href="step-by-step-tutorial.html#import-files-3"><strong><em>Import files</em></strong></a></li>
<li><a href="step-by-step-tutorial.html#define-bglr-1"><strong><em>Define BGLR</em></strong></a></li>
<li><a href="step-by-step-tutorial.html#inspect-results-2"><strong><em>Inspect results</em></strong></a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="4" data-path="formats.html"><a href="formats.html"><i class="fa fa-check"></i><b>4</b> Formats</a><ul>
<li class="chapter" data-level="4.1" data-path="formats.html"><a href="formats.html#input-formats"><i class="fa fa-check"></i><b>4.1</b> Input formats</a><ul>
<li class="chapter" data-level="4.1.1" data-path="formats.html"><a href="formats.html#genotype"><i class="fa fa-check"></i><b>4.1.1</b> Genotype</a></li>
<li class="chapter" data-level="4.1.2" data-path="formats.html"><a href="formats.html#phenotype"><i class="fa fa-check"></i><b>4.1.2</b> Phenotype</a></li>
<li class="chapter" data-level="4.1.3" data-path="formats.html"><a href="formats.html#map"><i class="fa fa-check"></i><b>4.1.3</b> Map</a></li>
<li class="chapter" data-level="4.1.4" data-path="formats.html"><a href="formats.html#covariates"><i class="fa fa-check"></i><b>4.1.4</b> Covariates</a></li>
<li class="chapter" data-level="4.1.5" data-path="formats.html"><a href="formats.html#kinship"><i class="fa fa-check"></i><b>4.1.5</b> Kinship</a></li>
</ul></li>
<li class="chapter" data-level="4.2" data-path="formats.html"><a href="formats.html#output-formats"><i class="fa fa-check"></i><b>4.2</b> Output formats</a><ul>
<li class="chapter" data-level="4.2.1" data-path="formats.html"><a href="formats.html#figures"><i class="fa fa-check"></i><b>4.2.1</b> Figures</a></li>
<li class="chapter" data-level="4.2.2" data-path="formats.html"><a href="formats.html#tabular-results"><i class="fa fa-check"></i><b>4.2.2</b> Tabular results</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="5" data-path="get-started-1.html"><a href="get-started-1.html"><i class="fa fa-check"></i><b>5</b> Get started</a><ul>
<li class="chapter" data-level="5.1" data-path="get-started-1.html"><a href="get-started-1.html#windows-users"><i class="fa fa-check"></i><b>5.1</b> Windows users</a></li>
<li class="chapter" data-level="5.2" data-path="get-started-1.html"><a href="get-started-1.html#mac-os-users"><i class="fa fa-check"></i><b>5.2</b> Mac OS users</a></li>
</ul></li>
<li class="chapter" data-level="6" data-path="interface-and-usage-1.html"><a href="interface-and-usage-1.html"><i class="fa fa-check"></i><b>6</b> Interface and Usage</a><ul>
<li class="chapter" data-level="6.1" data-path="interface-and-usage-1.html"><a href="interface-and-usage-1.html#workflow"><i class="fa fa-check"></i><b>6.1</b> Workflow</a><ul>
<li><a href="interface-and-usage-1.html#create-a-new-module"><strong><em>Create a new module</em></strong></a></li>
<li><a href="interface-and-usage-1.html#import-files-4"><strong><em>Import files</em></strong></a></li>
<li><a href="interface-and-usage-1.html#choose-an-analysis"><strong><em>Choose an analysis</em></strong></a></li>
<li><a href="interface-and-usage-1.html#config-and-run"><strong><em>Config and run</em></strong></a></li>
<li><a href="interface-and-usage-1.html#monitor-progress"><strong><em>Monitor progress</em></strong></a></li>
<li><a href="interface-and-usage-1.html#inspect-results-3"><strong><em>Inspect results</em></strong></a></li>
</ul></li>
<li class="chapter" data-level="6.2" data-path="interface-and-usage-1.html"><a href="interface-and-usage-1.html#global-settings"><i class="fa fa-check"></i><b>6.2</b> Global settings</a><ul>
<li class="chapter" data-level="6.2.1" data-path="interface-and-usage-1.html"><a href="interface-and-usage-1.html#working-directory"><i class="fa fa-check"></i><b>6.2.1</b> Working directory</a></li>
<li class="chapter" data-level="6.2.2" data-path="interface-and-usage-1.html"><a href="interface-and-usage-1.html#phenotype-1"><i class="fa fa-check"></i><b>6.2.2</b> Phenotype</a></li>
<li class="chapter" data-level="6.2.3" data-path="interface-and-usage-1.html"><a href="interface-and-usage-1.html#quality-control"><i class="fa fa-check"></i><b>6.2.3</b> Quality Control</a></li>
</ul></li>
<li class="chapter" data-level="6.3" data-path="interface-and-usage-1.html"><a href="interface-and-usage-1.html#gwas"><i class="fa fa-check"></i><b>6.3</b> GWAS</a><ul>
<li class="chapter" data-level="6.3.1" data-path="interface-and-usage-1.html"><a href="interface-and-usage-1.html#gapit"><i class="fa fa-check"></i><b>6.3.1</b> GAPIT</a></li>
<li class="chapter" data-level="6.3.2" data-path="interface-and-usage-1.html"><a href="interface-and-usage-1.html#farmcpu"><i class="fa fa-check"></i><b>6.3.2</b> FarmCPU</a></li>
<li class="chapter" data-level="6.3.3" data-path="interface-and-usage-1.html"><a href="interface-and-usage-1.html#plink-1"><i class="fa fa-check"></i><b>6.3.3</b> PLINK</a></li>
</ul></li>
<li class="chapter" data-level="6.4" data-path="interface-and-usage-1.html"><a href="interface-and-usage-1.html#gs"><i class="fa fa-check"></i><b>6.4</b> GS</a><ul>
<li class="chapter" data-level="6.4.1" data-path="interface-and-usage-1.html"><a href="interface-and-usage-1.html#gblup-and-rrblup"><i class="fa fa-check"></i><b>6.4.1</b> gBLUP and rrBLUP</a></li>
<li class="chapter" data-level="6.4.2" data-path="interface-and-usage-1.html"><a href="interface-and-usage-1.html#bglr"><i class="fa fa-check"></i><b>6.4.2</b> BGLR</a></li>
</ul></li>
<li class="chapter" data-level="6.5" data-path="interface-and-usage-1.html"><a href="interface-and-usage-1.html#gwas-assisted-gs"><i class="fa fa-check"></i><b>6.5</b> GWAS-assisted GS</a></li>
<li class="chapter" data-level="6.6" data-path="interface-and-usage-1.html"><a href="interface-and-usage-1.html#format-converter"><i class="fa fa-check"></i><b>6.6</b> Format converter</a></li>
<li class="chapter" data-level="6.7" data-path="interface-and-usage-1.html"><a href="interface-and-usage-1.html#multi-thread-is-supported"><i class="fa fa-check"></i><b>6.7</b> Multi-thread is supported</a></li>
</ul></li>
</ul>

      </nav>
    </div>

    <div class="book-body">
      <div class="body-inner">
        <div class="book-header" role="navigation">
          <h1>
            <i class="fa fa-circle-o-notch fa-spin"></i><a href="./">GRID: Greenfield Image Decoder</a>
          </h1>
        </div>

        <div class="page-wrapper" tabindex="-1" role="main">
          <div class="page-inner">

            <section class="normal" id="section-">
<div id="interface-and-usage-1" class="section level1">
<h1><span class="header-section-number">6</span> Interface and Usage</h1>
<p>In this chapter, we’ll walk through the basic workflow in iPat and how users can define their analyses with different tools. Also, other miscellaneous features will be addressed in the later section of this chapter.</p>
<div id="workflow" class="section level2">
<h2><span class="header-section-number">6.1</span> Workflow</h2>
<p>To perform GWAS, GS, and GWAS-assisted GS in iPat, following six steps are demonstrated for users to follow:</p>
<div id="create-a-new-module" class="section level3 unnumbered">
<h3><strong><em>Create a new module</em></strong></h3>
<p>Every single analysis is defined by a ‘module’, which can be created by double-click anywhere in the screen and opened by right-click on a gear-shaped icon which is newly created. It’s noted that there will always be an automatically created module at the very beginning when users launch iPat, so users only need to right-click on the gear-shaped icon to start their first analysis.</p>
<p align="center">
<img src = "md/2.1-CreateMod.gif" width = 900>
</p>
<p align="center">
Fig. Right click to open a module
</p>
</div>
<div id="import-files-4" class="section level3 unnumbered">
<h3><strong><em>Import files</em></strong></h3>
<p>Users can drag-and-drop their files to import them into iPat. Each filetype has its own slot, and each analysis have slightly different requirements for the imported files. Please follow the instruction in section <a href="interface-and-usage-1.html#gwas">2.3</a>, <a href="interface-and-usage-1.html#gs">2.4</a>, <a href="interface-and-usage-1.html#format-converter">2.5</a> to learn for further information.</p>
<p align="center">
<img src = "md/2.1-Import.gif" width = 900>
</p>
<p align="center">
Fig. Drag-and-drop files to import
</p>
</div>
<div id="choose-an-analysis" class="section level3 unnumbered">
<h3><strong><em>Choose an analysis</em></strong></h3>
<p>Currently there’re four analyses available in iPat. Once users done importing all the required files, they can click a desired analysis to enter the next panel for further configuration.</p>
<p align="center">
<img src = "md/2.1-ChooseAna.gif" width = 900>
</p>
<p align="center">
Fig. Choose an analysis for further configuration
</p>
</div>
<div id="config-and-run" class="section level3 unnumbered">
<h3><strong><em>Config and run</em></strong></h3>
<p>Usually there’re two parts in this panel. In the upper part, users can specify the working directory, which phenotype to be analyzed, and quality controls on genetic markers. This panel can be globally defined regardless chosen analysis (<a href="interface-and-usage-1.html#global-settings">section -2.2</a>). Whereas the sub-panel in the bottom part is tool-specific, this area allows users to define the chosen tool in details. Please read section <a href="interface-and-usage-1.html#gwas">2.3</a>, <a href="interface-and-usage-1.html#gs">2.4</a>, <a href="interface-and-usage-1.html#format-converter">2.5</a> for further information. The analysis will be launched when ‘run’ button in the right-corner is clicked.</p>
<p align="center">
<img src = "md/2.1-Config.gif" width = 900>
</p>
<p align="center">
Fig. Detailed configuration for the analysis
</p>
</div>
<div id="monitor-progress" class="section level3 unnumbered">
<h3><strong><em>Monitor progress</em></strong></h3>
<p>Users can hover the pointer on the running module to monitor its progress, or to click on the module to lock the progress window, which can prevent it from disappearing when the pointer is no longer hovered on the module.</p>
<p align="center">
<img src = "md/2.1-Monitor.gif" width = 900>
</p>
<p align="center">
Fig. Hover the pointer to monitor the progress
</p>
</div>
<div id="inspect-results-3" class="section level3 unnumbered">
<h3><strong><em>Inspect results</em></strong></h3>
<p>Once the analysis is done, a green dot will present at the upper-left corner of the module. Users can open the working directory and find all the output files by double-clicking on the module.</p>
<p align="center">
<img src = "md/2.1-Result.gif" width = 900>
</p>
<p align="center">
Fig. Double click to open the working directory
</p>
</div>
</div>
<div id="global-settings" class="section level2">
<h2><span class="header-section-number">6.2</span> Global settings</h2>
<div id="working-directory" class="section level3">
<h3><span class="header-section-number">6.2.1</span> Working directory</h3>
<p>Users can specify the working directory and module names in this panel. Module names will be presented in the output, which can serve as a label for users to distinguish different analyses.</p>
<p align="center">
<img src = "md/2.2-WD.png" width = 900>
</p>
<p align="center">
Fig. Panel for users to define working directory
</p>
</div>
<div id="phenotype-1" class="section level3">
<h3><span class="header-section-number">6.2.2</span> Phenotype</h3>
<p>Sometimes users may have several traits in one phenotype file, but not all of them are traits of interest. To deal with this concern, in this panel, traits can be selected and excluded for the following analysis. Click ‘–&gt;’ button to exclude traits that are not the interest of this analysis, or click ‘&lt;–’ button to rescue back traits that are mistakenly excluded.</p>
<p align="center">
<img src = "md/2.2-Phenotype.png" width = 900>
</p>
<p align="center">
Fig. Panel for users to select traits of interest
</p>
</div>
<div id="quality-control" class="section level3">
<h3><span class="header-section-number">6.2.3</span> Quality Control</h3>
<p>Two ways of quality controls (QC) are available in this panel, both ways will be applied to genetic markers. ‘By missing rate’ will filter out any marker that has missing rate (i.e., 1 - calling rate) <strong>higher</strong> than the assigned value. ‘By MAF’ will filter out markers with minor allele frequency <strong>lower</strong> than the assigned value.</p>
<p align="center">
<img src = "md/2.2-QC.png" width = 900>
</p>
<p align="center">
Fig. Sliders for performing quality controls
</p>
</div>
</div>
<div id="gwas" class="section level2">
<h2><span class="header-section-number">6.3</span> GWAS</h2>
<p>It’s required to provide <a href="formats.html#genotype">genotype</a>, <a href="formats.html#map">map</a> (if applicable), and <a href="formats.html#phenotype">phenotype</a> files to run GWAS in iPat. <a href="formats.html#covariates">Covariates</a> and <a href="formats.html#kinship">kinship</a> matrix are optional. User-provided <a href="formats.html#covariates">covariates</a> will serve as fixed effects in the GWAS model. Often time users are not recommended to provide a <a href="formats.html#kinship">kinship</a> matrix, since those implemented tools will generate one automatically and user-define <a href="formats.html#kinship">kinship</a> is used only in ‘MLM’ and ‘CMLM’ from GAPIT.</p>
<p>Output files from GWAS include:</p>
<ul>
<li>Figures: <a href="formats.html#manhattan-plot">Manhattan plot</a>, <a href="formats.html#scatter-plot-q-q-plot">Q-Q plot</a>, <a href="formats.html#histogram-heterozygosity-distribution">Heterozygosity distribution</a>, and <a href="formats.html#phenotype-overview">Phenotype overview</a></li>
<li>Tables: <a href="formats.html#gwas-report">GWAS report</a></li>
</ul>
<div id="gapit" class="section level3">
<h3><span class="header-section-number">6.3.1</span> GAPIT</h3>
<table class="table" style="width: auto !important; margin-left: auto; margin-right: auto;">
<caption>
<span id="tab:unnamed-chunk-14">Table 6.1: </span>Arguments for GAPIT
</caption>
<thead>
<tr>
<th style="text-align:left;">
Arguments
</th>
<th style="text-align:left;">
Default
</th>
<th style="text-align:left;">
Options
</th>
<th style="text-align:left;">
Description
</th>
</tr>
</thead>
<tbody>
<tr>
<td style="text-align:left;">
Model
</td>
<td style="text-align:left;">
GLM
</td>
<td style="text-align:left;">
GLM, MLM, CMLM, FarmCPU
</td>
<td style="text-align:left;width: 20em; ">
Implemented models: general linear model (GLM), mixed linear model (MLM), compressed mixed linear model (CMLM), and FarmCPU.
</td>
</tr>
<tr>
<td style="text-align:left;">
PC
</td>
<td style="text-align:left;">
3
</td>
<td style="text-align:left;">
Ranges between 3 to 10
</td>
<td style="text-align:left;width: 20em; ">
Principal components to control spurious signals caused by population stratifications
</td>
</tr>
</tbody>
</table>
<p align="center">
<img src = "md/2.3-GAPIT.png" width = 500>
</p>
<p align="center">
Fig. Configuration for GAPIT
</p>
<p><strong>Reference</strong></p>
<ul>
<li>Tang,Y. et al. (2016) GAPIT Version 2: An Enhanced Integrated Tool for Genomic Association and Prediction. Plant J., 9.</li>
</ul>
</div>
<div id="farmcpu" class="section level3">
<h3><span class="header-section-number">6.3.2</span> FarmCPU</h3>
<table class="table" style="width: auto !important; margin-left: auto; margin-right: auto;">
<caption>
<span id="tab:unnamed-chunk-15">Table 6.2: </span>Arguments for FarmCPU
</caption>
<thead>
<tr>
<th style="text-align:left;">
Arguments
</th>
<th style="text-align:left;">
Default
</th>
<th style="text-align:left;">
Options
</th>
<th style="text-align:left;">
Description
</th>
</tr>
</thead>
<tbody>
<tr>
<td style="text-align:left;">
Method bin
</td>
<td style="text-align:left;">
optimum
</td>
<td style="text-align:left;">
static, optimum
</td>
<td style="text-align:left;width: 20em; ">
</td>
</tr>
<tr>
<td style="text-align:left;">
maxLoop
</td>
<td style="text-align:left;">
10
</td>
<td style="text-align:left;">
Ranges between 1 to 10
</td>
<td style="text-align:left;width: 20em; ">
Number of iterations for FarmCPU to detect QTNs
</td>
</tr>
</tbody>
</table>
<p align="center">
<img src = "md/2.3-FarmCPU.png" width = 500>
</p>
<p align="center">
Fig. Configuration for FarmCPU
</p>
<p><strong>Reference</strong></p>
<ul>
<li>Liu,X. et al. (2016) Iterative Usage of Fixed and Random Effect Models for Powerful and Efficient Genome-Wide Association Studies. PLoS Genet., 12 , e1005767</li>
</ul>
</div>
<div id="plink-1" class="section level3">
<h3><span class="header-section-number">6.3.3</span> PLINK</h3>
<table class="table" style="width: auto !important; margin-left: auto; margin-right: auto;">
<caption>
<span id="tab:unnamed-chunk-16">Table 6.3: </span>Arguments for PLINK
</caption>
<thead>
<tr>
<th style="text-align:left;">
Arguments
</th>
<th style="text-align:left;">
Default
</th>
<th style="text-align:left;">
Options
</th>
<th style="text-align:left;">
Description
</th>
</tr>
</thead>
<tbody>
<tr>
<td style="text-align:left;">
C.I.
</td>
<td style="text-align:left;">
0.95
</td>
<td style="text-align:left;">
0.95, 0.975, 0.995
</td>
<td style="text-align:left;width: 20em; ">
Confident interval for testing marker effects
</td>
</tr>
<tr>
<td style="text-align:left;">
Method
</td>
<td style="text-align:left;">
GLM
</td>
<td style="text-align:left;">
GLM, Logistic Regression
</td>
<td style="text-align:left;width: 20em; ">
Implemented models
</td>
</tr>
</tbody>
</table>
<p align="center">
<img src = "md/2.3-PLINK.png" width = 500>
</p>
<p align="center">
Fig. Configuration for PLINK
</p>
<p><strong>Reference</strong></p>
<ul>
<li>Purcell,S. et al. (2007) PLINK: A Tool Set for Whole-Genome Association and Population-Based Linkage Analyses. Am J Hum Genet, 81, 559–575.</li>
</ul>
</div>
</div>
<div id="gs" class="section level2">
<h2><span class="header-section-number">6.4</span> GS</h2>
<p>It’s required to provide <a href="formats.html#genotype">genotype</a> and <a href="formats.html#phenotype">phenotype</a> files to run GS in iPat. User-provided <a href="formats.html#covariates">covariates</a> will serve as fixed effects in the GS model, but none of GS methods will take user-defined <a href="formats.html#kinship">kinship</a> in iPat.</p>
<p>In iPat, users can perform two types of GS. If the ‘validation’ option is unchecked, iPat would estimate genetic estimated breeding values (GEBV) for samples in the whole dataset based on the phenotyped samples. Whereas when users check the option ‘Validation on accuracy’, iPat will split whole dataset into training and testing samples in order to verify the capability of trained GS model. For this manner, validation accuracies will be reported instead of GEBVs.</p>
<table class="table" style="width: auto !important; margin-left: auto; margin-right: auto;">
<caption>
<span id="tab:unnamed-chunk-17">Table 6.4: </span>Arguments for GS
</caption>
<thead>
<tr>
<th style="text-align:left;">
Arguments
</th>
<th style="text-align:left;">
Default
</th>
<th style="text-align:left;">
Options
</th>
<th style="text-align:left;">
Description
</th>
</tr>
</thead>
<tbody>
<tr>
<td style="text-align:left;">
Validation on accuracy
</td>
<td style="text-align:left;">
unchecked
</td>
<td style="text-align:left;">
checked, unchecked
</td>
<td style="text-align:left;width: 20em; ">
If k-fold validation is performed in GS
</td>
</tr>
<tr>
<td style="text-align:left;">
Folds
</td>
<td style="text-align:left;">
3
</td>
<td style="text-align:left;">
3, 5, 10
</td>
<td style="text-align:left;width: 20em; ">
K value for K-fold validation. For example, if k = 5, 20% samples will be tested as accuracy validation.
</td>
</tr>
<tr>
<td style="text-align:left;">
Iteration
</td>
<td style="text-align:left;">
10
</td>
<td style="text-align:left;">
Range from 10 to 100
</td>
<td style="text-align:left;width: 20em; ">
Number of iterations. Each iteration will split whole dataset into training part and testing part based on the specified k value for K-fold validation.
</td>
</tr>
</tbody>
</table>
<p align="center">
<img src = "md/2.4-Validation.png" width = 500>
</p>
<p>Output files from GS include:</p>
<ul>
<li>Figures: <a href="formats.html#histogram-gebv-distribution">GEBV distribution</a>, <a href="#density-function-accuracies-for-gs-validation">Validation accuracies</a>, <a href="#scatter-plot-gs-predictions-against-observation">Predicted versus observed values</a>, <a href="formats.html#histogram-heterozygosity-distribution">Heterozygosity distribution</a>, and <a href="formats.html#phenotype-overview">Phenotype overview</a>.</li>
<li>Tables: <a href="formats.html#gs-gebv">GEBV report</a>, <a href="formats.html#gs-marker-effects">Marker effects</a>, <a href="formats.html#gs-validation-results">Validation results</a>, and <a href="formats.html#gs-predicted-values">Predicted values</a></li>
</ul>
<div id="gblup-and-rrblup" class="section level3">
<h3><span class="header-section-number">6.4.1</span> gBLUP and rrBLUP</h3>
<p>There’s no specific arguments available for both of these methods.</p>
<p><strong>Reference</strong></p>
<ul>
<li>Endelman,J. (2011) Ridge regression and other kernels for genomic selection in the R package rrBLUP. Plant Genome, 4, 250–255.</li>
</ul>
</div>
<div id="bglr" class="section level3">
<h3><span class="header-section-number">6.4.2</span> BGLR</h3>
<table class="table" style="width: auto !important; margin-left: auto; margin-right: auto;">
<caption>
<span id="tab:unnamed-chunk-18">Table 6.5: </span>Arguments for BGLR
</caption>
<thead>
<tr>
<th style="text-align:left;">
Arguments
</th>
<th style="text-align:left;">
Default
</th>
<th style="text-align:left;">
Options
</th>
<th style="text-align:left;">
Description
</th>
</tr>
</thead>
<tbody>
<tr>
<td style="text-align:left;">
Model for predictors
</td>
<td style="text-align:left;">
BRR
</td>
<td style="text-align:left;">
BRR, Bayes A, Bayes B, Bayes C, BL
</td>
<td style="text-align:left;width: 20em; ">
Model to estimate marker effects
</td>
</tr>
<tr>
<td style="text-align:left;">
nIter
</td>
<td style="text-align:left;">
5k
</td>
<td style="text-align:left;">
Range from 1k to 100k
</td>
<td style="text-align:left;width: 20em; ">
Number of iteration
</td>
</tr>
<tr>
<td style="text-align:left;">
burnIn
</td>
<td style="text-align:left;">
500
</td>
<td style="text-align:left;">
Range from 200 to 10k
</td>
<td style="text-align:left;width: 20em; ">
Burn-in values
</td>
</tr>
</tbody>
</table>
<p align="center">
<img src = "md/2.4-BGLR.png" width = 500>
</p>
<p><strong>Reference</strong></p>
<ul>
<li>Pérez, Paulino. et al. (2014) Genome-wide regression and prediction with the BGLR statistical package. Genetics, 198, 483-495.</li>
</ul>
</div>
</div>
<div id="gwas-assisted-gs" class="section level2">
<h2><span class="header-section-number">6.5</span> GWAS-assisted GS</h2>
<p>It’s a hybrid analysis that includes GWAS reported markers as covariates (fixed effects) in the GS model, whichever tested marker that has an association signal more significant than Bonferroni cutoff (<em>p-value</em> &lt; 0.05 / number of markers) will be considered as such covariates. The workflow for GWAS-assisted GS in iPat is basically the same as performing GWAS and GS alone, please follow the <a href="step-by-step-tutorial.html#gwas-assisted-gs-in-rrblup">tutorial</a> for details.</p>
</div>
<div id="format-converter" class="section level2">
<h2><span class="header-section-number">6.6</span> Format converter</h2>
<p>iPat not only provides a format-free environment for genomic studies, it also allows users to utilize the build-in converter to convert files for other further purposes. By providing a genotype file and a map file (if applicable) that meet a certain format, iPat can convert them into numerical-encoded genotypes or PLINK format that is compatible to other software.</p>
<p align="center">
<img src = "md/2.6-Converter.png" width = 900>
</p>
<p>Other than specifying requested formats and arguments for quality controls, users can decide whether they would like to have missing values filled as heterozygous genotypes. Also, the argument “Batch size” determines how many samples are read into the memory at a time. The larger the batch size is, the faster the conversion would be. However, for those computing units with limited memory resources, it’s highly recommended to start from a smaller number such as 64 or 32. Large batch size will crash the program easily when dealing with a large number of markers.</p>
<table class="table" style="width: auto !important; margin-left: auto; margin-right: auto;">
<caption>
<span id="tab:unnamed-chunk-19">Table 6.6: </span>Arguments for iPat converter
</caption>
<thead>
<tr>
<th style="text-align:left;">
Arguments
</th>
<th style="text-align:left;">
Default
</th>
<th style="text-align:left;">
Options
</th>
</tr>
</thead>
<tbody>
<tr>
<td style="text-align:left;">
Input formats
</td>
<td style="text-align:left;">
None
</td>
<td style="text-align:left;">
Numerical, Hapmap, VCF, PLINK
</td>
</tr>
<tr>
<td style="text-align:left;">
Output formats
</td>
<td style="text-align:left;">
None
</td>
<td style="text-align:left;">
Numerical, PLINK
</td>
</tr>
<tr>
<td style="text-align:left;">
By missing rate
</td>
<td style="text-align:left;">
0.2
</td>
<td style="text-align:left;">
0, 0.05, 0.1, 0.2, 0.5
</td>
</tr>
<tr>
<td style="text-align:left;">
By MAF
</td>
<td style="text-align:left;">
0.05
</td>
<td style="text-align:left;">
0, 0.01, 0.05, 0.1, 0.2
</td>
</tr>
<tr>
<td style="text-align:left;">
Fill NA
</td>
<td style="text-align:left;">
unchecked
</td>
<td style="text-align:left;">
checked, unchecked
</td>
</tr>
<tr>
<td style="text-align:left;">
Batch size
</td>
<td style="text-align:left;">
64
</td>
<td style="text-align:left;">
32, 64, 128, 256, 512
</td>
</tr>
</tbody>
</table>
</div>
<div id="multi-thread-is-supported" class="section level2">
<h2><span class="header-section-number">6.7</span> Multi-thread is supported</h2>

</div>
</div>
            </section>

          </div>
        </div>
      </div>
<a href="get-started-1.html" class="navigation navigation-prev navigation-unique" aria-label="Previous page"><i class="fa fa-angle-left"></i></a>

    </div>
  </div>
<script src="libs/gitbook-2.6.7/js/app.min.js"></script>
<script src="libs/gitbook-2.6.7/js/lunr.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-search.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-sharing.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-fontsettings.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-bookdown.js"></script>
<script src="libs/gitbook-2.6.7/js/jquery.highlight.js"></script>
<script>
gitbook.require(["gitbook"], function(gitbook) {
gitbook.start({
"sharing": {
"github": false,
"facebook": true,
"twitter": true,
"google": false,
"linkedin": false,
"weibo": false,
"instapaper": false,
"vk": false,
"all": ["facebook", "google", "twitter", "linkedin", "weibo", "instapaper"]
},
"fontsettings": {
"theme": "white",
"family": "sans",
"size": 2
},
"edit": {
"link": null,
"text": null
},
"history": {
"link": null,
"text": null
},
"download": null,
"toc": {
"collapse": "subsection"
}
});
});
</script>

</body>

</html>
